File: 1929-concatenation-of-array/1929-concatenation-of-array.java
class Solution {
    public int[] getConcatenation(int[] nums) {
        return addAll(nums, nums);
    }
    
    int[] addAll(int[] array1, int[] array2) {
        if (array1 == null) {
            return clone(array2);
        } else if (array2 == null) {
            return clone(array1);
        } else {
            int[] joinedArray = new int[array1.length + array2.length];
            System.arraycopy(array1, 0, joinedArray, 0, array1.length);
            System.arraycopy(array2, 0, joinedArray, array1.length, array2.length);
            return joinedArray;
        }
    }

    int[] clone(int[] array) {
        return array == null ? null : (int[])array.clone();
    }
}

File: 1929-concatenation-of-array/1929-concatenation-of-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var getConcatenation = function(nums) {
    let res=[]
    for(let i=0;i<nums.length;i++){
        res[i]=nums[i]
        res[i+nums.length]=nums[i]
    }
    return res
};